saml/response.go

```diff
@@ -245,3 +254,21 @@ func parsePEMCertificate(cert []byte) (*x509.Certificate, error) {
 
 	return x509.ParseCertificate(block.Bytes)
 }
+
+func validateSignature(response *types.Response, op string) error {
+	// validate child object assertions
+	for _, assert := range response.Assertions {
```

Comment:
gosaml2 requires either the response or the assertions to be signed. It doesn't require both. 

My qustion would be if we need to sign both because signature inheritance applies (if the parent object is signed, the child object is included)?
Having both objects signed makes sense in some instances where the service provider could retrieve the assertion outside of a SAML Response, but that's not the case here.

Reply:
I did some further digging on this. Most SAML IDPs (e.g. Entra ID and Okta) do provide the option to sign both: `Response` and `Assertions`. However, it looks like there are indeed some IDPs that do not provide the option, such as `Auth0`. This is from their application settings:

```
signResponse (bool): Whether or not the SAML Response should be signed. By default the SAML Assertion will be signed, but not the SAML Response. If true, SAML Response will be signed instead of SAML Assertion.
```

On your edit above in the description:
> After few major version, the vault variable will be removed and user will always have to sign both.

Making both a requirement might lead to incompatibilities with some IDPs (like Auth0 for example). So I wonder if supporting all 3 options would be the ideal case:
- Either `Response` or `Assertion` has to be signed
- `Response` has to be signed
- Both have to be signed.

Reply:
Changes done to add separate options for all signature validation. Really appreciate for you finding the info on Auth0 as IDP.

Reply:
Also btw the options i kept are: 
- assertion is signed
- response is signed
- both are signed
Did not include the `either` option as i thought user should makes a known decision relative to their IDP. Let me know if you think otherwise.

Reply:
Hey @himran92 sorry for the late response and thx a lot for addressing my comments 🙇 

Sounds great. The options makes totally sense. 

The default would still be `either`, correct?

Reply:
I am catching up after vacations :). 

yes correct. 
The default where neither of these options is provided is still `either`. 
This is given by these [unit tests](https://github.com/hashicorp/cap/blob/d857fbab8bb1108868cdd6b03433293f72304809/saml/response_test.go#L68-L81).